src/library/scala/collection/IndexedSeqLike.scala

```diff
@@ -87,7 +87,11 @@ trait IndexedSeqLike[+A, +Repr] extends Any with SeqLike[A, Repr] {
   }
 
   override /*IterableLike*/
-  def iterator: Iterator[A] = new Elements(0, length)
+  def iterator: Iterator[A] = {
+    val len = length
+    if (len == 0) Iterator.empty
+    else new Elements(0, length)
```

Comment:
len? I am if/else averse and would have `length match case 0 case len`

Reply:
If/else is objectively a better choice than match; there's no matching happening and it's more compilation efficient. Match doesn't exist to make if/else obsolete.

Reply:
Match subsumes if/else, IMO.  And vertical alignment of arrows... ❤️